### PR TITLE
Initial commit for UI/UX Kit

### PR DESCRIPTION
**Pull request by kanishk2012**
Initial commit for UI/UX Kit for using together with M800 Communication SDK Suite

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+source 'https://github.com/CocoaPods/Specs.git'
+source 'http://gerrit-mobile.dev.maaii.com/cocoapods' 
+
+platform :ios, '9.0'
+
+target 'M800UIKit' do
+
+  use_frameworks!
+
+  pod 'M800SDKCore', '~> 3.2.0-alpha'
```
Comment (dgutzeit) [inline Podfile]: I don't like this. we should not depend on an alpha SDK

Comment (kanishk2012) [reply Podfile]: I am updating to the exact versions without alpha.

File: M800UIKit/Modules/Contacts/ViewModel/M800UserViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        profileVideoURL = profile.profileVideoURL
+        profileImageURL = profile.profileImageURL
+        profileCoverImageURL = profile.profileCoverImageURL
+        profileVideoThumbURL = profile.profileVideoThumbURL
+        profileImageThumbnailURL = profile.profileImageThumbURL
+    }
+
+    private func getBackupName(jid: String) -> String {
+
+        guard !jid.isEmpty, let range = jid.range(of: "@") else { return "Anonymous" }
```
[the remark this answers is not in the document]
Comment (kanishk2012) [reply M800UIKit/Modules/Contacts/ViewModel/M800UserViewModel.swift]: Yes, should be. I will update in next patchset.

File: M800UIKit/Modules/Contacts/View/M800ProfileHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        imageView.contentMode = .scaleAspectFill
+
+        if let path = path, let url = URL(string: path) {
+
+            M800ImageStorage.shared.loadImage(url: url, closure: { [weak self] (image) in
+                self?.imageView?.image = image
+            })
+
+        } else {
+            // The source should not be replaced here...
```
[the remark this answers is not in the document]
Comment (kanishk2012) [reply M800UIKit/Modules/Contacts/View/M800ProfileHeaderView.swift]: Yes, we are trying to make sure the default source image path is used for default image "user_profile_cover_default"

**Pull request merged**